runtime/sema/check_interface_declaration.go

```diff
@@ -156,21 +156,26 @@ func (checker *Checker) VisitInterfaceDeclaration(declaration *ast.InterfaceDecl
 	}
 
 	for _, nestedComposite := range declaration.Members.Composites() {
-		// Non-event composite declarations nested in interface declarations are type requirements,
-		// i.e. they should be checked like interfaces
-
+		// only event types may be declared in interfaces
 		if nestedComposite.Kind() != common.CompositeKindEvent {
-			checker.visitCompositeLikeDeclaration(nestedComposite, kind)
+			checker.report(&InvalidNestedDeclarationError{
```

Comment:
Looks like the error reporting for invalid nested declarations is reported here. Maybe do nothing here, remove the filtering above (see comment above), and add a comment here instead noting that this case is a NO-OP, and an error is reported for all invalid nestings elsewhere.